code/Controllers/ModelAsController.php

```diff
@@ -33,11 +33,9 @@ class ModelAsController extends Controller implements NestedController
      * Get the appropriate {@link ContentController} for handling a {@link SiteTree} object, link it to the object and
      * return it.
      *
-     * @param SiteTree $sitetree
      * @param string $action
-     * @return ContentController
      */
-    public static function controller_for(SiteTree $sitetree, $action = null)
+    public static function controller_for(SiteTree $sitetree, $action = null): ContentController
```

Comment:
If anyone was using `ModelAsController` to get their controller for a non-sitetree this will break - we need to mention it on the changelog (or just not make the change - I don't know that it's valuable to make this change but I'll leave it to you to decide which to do)

Reply:
Note: If we do revert this change then https://github.com/silverstripe/silverstripe-cms/pull/2788#discussion_r995163934 will need to be looked at again, but I've marked that resolved for now.

Reply:
> get their controller for a non-sitetree

There was already strong typing for the `SiteTree $sitetree` param

I do not think this needs to be called out in the changelog